src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java

```diff
@@ -114,44 +104,16 @@ public <T> T callForObject(String functionName, List<?> parameters, Class<T> ent
         );
     }
 
-    private MessagePackMapper getDefaultComplexTypesMapper() {
-        // FIXME: https://github.com/tarantool/cartridge-java/issues/166
-        // return tarantoolClient.getConfig().getMessagePackMapper().copy();
-        return DefaultMessagePackMapperFactory
-                .getInstance()
-                .defaultComplexTypesMapper();
-    }
-
-    private void registerTupleResultMapper(MessagePackMapper customMapper,
-                                           Optional<TarantoolSpaceMetadata> spaceMetadata) {
-        customMapper.registerValueConverter(
-                ValueType.ARRAY,
-                TarantoolTuple.class,
-                new TarantoolAutoResultConverter<>(
-                        new TarantoolTupleConverter(customMapper,
-                                spaceMetadata.orElse(null)),
-                        spaceMetadata.orElse(null)));
-        customMapper.registerValueConverter(
-                ValueType.MAP,
-                TarantoolTuple.class,
-                new TarantoolAutoResultConverter<>(
-                        new TarantoolTupleConverter(customMapper,
-                                spaceMetadata.orElse(null)),
-                        spaceMetadata.orElse(null)));
-    }
-
     private <T> CallResultMapper<T, SingleValueCallResult<T>>
-    withDefaultSingleValueMapper(MessagePackMapper customMapper,
-                                 Class<T> entityClass) {
-        return mapperFactoryFactory
-                .getDefaultSingleValueMapper(customMapper, entityClass);
-    }
-
-    private <T> CallResultMapper<T, SingleValueCallResult<T>>
-    getAutoResultMapper(Class<T> entityClass, Optional<TarantoolSpaceMetadata> spaceMetadata) {
-        MessagePackMapper customMapper = getDefaultComplexTypesMapper();
-        registerTupleResultMapper(customMapper, spaceMetadata);
-        return withDefaultSingleValueMapper(customMapper, entityClass);
+    getAutoResultMapper(Optional<TarantoolSpaceMetadata> spaceMetadata) {
```

Comment:
Looks like that we could combine this builder sequence with the same one above and make a factory method that accepts a single mapper (that's the only difference it seems).